test/e2e/framework/pod/create.go

```diff
@@ -223,33 +215,42 @@ func MakePodSpec(podConfig *Config) *v1.PodSpec {
 		podSpec.SecurityContext.FSGroupChangePolicy = podConfig.PodFSGroupChangePolicy
 	}
 
+	setVolumes(podSpec, podConfig.PVCs, podConfig.InlineVolumeSources, podConfig.PVCsReadOnly)
+	SetNodeSelection(podSpec, podConfig.NodeSelection)
+	return podSpec
+}
+
+func setVolumes(podSpec *v1.PodSpec, pvcs []*v1.PersistentVolumeClaim, inlineVolumeSources []*v1.VolumeSource, pvcsReadOnly bool) {
```

Comment:
`MakePodSec` was setting block and inline volumes correctly, so instead of copying the code to `MakePod`, I moved it to a separate function (`setVolumes`) that can be called from `MakePod` and `MakePodSec`.